### PR TITLE
add service layer as well the handler layer test cases

### PR DESCRIPTION
**Pull request by purisaurabh**
This PR includes test cases of service layer as well as the handler layer

### CODE REVIEW

File: internal/api/tests/user_email_handler_test.go [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+
+func TestHandlerTestSuite(t *testing.T) {
+	suite.Run(t, new(HandlerTestSuite))
+}
+
+func (s *HandlerTestSuite) SetupTest() {
+	s.service = &mocks.Service{}
+}
+
+func (suite *HandlerTestSuite) TearDownSuite() {
```
Comment (OmkeshJosh) [inline internal/api/tests/user_email_handler_test.go]: use `(s *HandlerTestSuite)`

File: internal/api/tests/achievement_handler_test.go [hunk 2]
```diff
@@ -30,25 +31,62 @@ func TestCreateAchievementHandler(t *testing.T) {
 		input              string
 		setup              func(mockSvc *mocks.Service)
 		expectedStatusCode int
+		expectedResponse   string
 	}{
 		{
```
Comment (OmkeshJosh) [inline internal/api/tests/achievement_handler_test.go:38]: Please add fail test case for
- GetParamsByID
- GetUserIDFromContext

File: internal/api/tests/achievement_handler_test.go [hunk 3]
```diff
@@ -60,125 +98,143 @@ func TestCreateAchievementHandler(t *testing.T) {
...
+	updateAchievementHandler := handler.UpdateAchievementHandler(context.Background(), achSvc)
 	tests := []struct {
 		name               string
 		input              string
 		setup              func(mockSvc *mocks.Service)
 		expectedStatusCode int
+		expectedResponse   string
 	}{
 		{
 			name: "Success_for_achievement_update",
```
Comment (OmkeshJosh) [inline internal/api/tests/achievement_handler_test.go:190]: Add test case for
- Fail GetMultipleParams
- Fail GetUserIDFromContext

File: internal/api/tests/achievement_handler_test.go [hunk 4]
```diff
@@ -199,16 +255,17 @@ func TestListAchievementsHandler(t *testing.T) {
 		mockDecodeRequest  func()
 		MockSvc            func(mockSvc *mocks.Service)
 		expectedStatusCode int
+		expectedResponse   string
 	}{
 		{
```
Comment (OmkeshJosh) [inline internal/api/tests/achievement_handler_test.go:328]: Add test case for
- fail DecodeAchievementRequest and GetParamsByID

**Comment by OmkeshJosh** [comment]
1. Please fix golint issue : 

<pre>golint ./...
internal/api/tests/user_email_handler_test.go:31:1: receiver name suite should be consistent with previous receiver name s for HandlerTestSuite
internal/pkg/helpers/helpers.go:141:1: exported function ConvertToLowerCase should have comment or be unexported
internal/repository/user_login_repository.go:83:2: var checkSql should be checkSQL </pre>





2. Add missing test cases by using following commands :  

<pre>go test -v -coverprofile cover.out ./...
go tool cover -html cover.out -o cover.html
</pre>

File: internal/api/tests/user_email_handler_test.go (hunk 1 above)
Comment (purisaurabh) [reply internal/api/tests/user_email_handler_test.go]: done

File: internal/api/tests/achievement_handler_test.go (hunk 2 above)
Comment (purisaurabh) [reply internal/api/tests/achievement_handler_test.go:38]: Done

File: internal/api/tests/achievement_handler_test.go (hunk 4 above)
Comment (purisaurabh) [reply internal/api/tests/achievement_handler_test.go:328]: Done

File: internal/api/tests/achievement_handler_test.go (hunk 3 above)
Comment (purisaurabh) [reply internal/api/tests/achievement_handler_test.go:190]: Done

**Pull request merged**